### PR TITLE
openssl3: update to 3.3.1 and 3.1.6

### PR DESCRIPTION
**Pull request by herbygillot**
- remove patches that are now already part of upstream

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with ~`sudo port -vst install`~ `sudo port -d destroot`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by mascguy** [comment]
> @fhgwright @mascguy @kencu @catap Could anyone say why exactly we keep older systems at 3.1.x and if the issue is still unfixed, has anyone reported it to OpenSSL upstream?

You should review the commit history of this port, as well as old Trac tickets.

**Review by neverpanic — APPROVED** [review]

**Comment by neverpanic** [comment]
> @fhgwright @mascguy @kencu @catap Could anyone say why exactly we keep older systems at 3.1.x and if the issue is still unfixed, has anyone reported it to OpenSSL upstream?

The people that are experiencing the issue are either unable or don't have the time to actually debug it, as it only affects `+universal` on old macOS. It may be caused by a buggy compiler.

Due to the limited impact (old OS & universal only) I'm planning to just drop the 3.1.x branch as soon as it is end-of-life:

> Version 3.1 will be supported until 2025-03-14
> https://www.openssl.org/policies/releasestrat.html

**Comment by barracuda156** [comment]
@neverpanic Well, please wait, let us verify if the issue is still there or not. (And I am pretty sure that before March 2025 we can fix it with OpenSSL upstream, if it is still present.)

**Comment by neverpanic** [comment]
> @neverpanic Well, please wait, let us verify if the issue is still there or not. (And I am pretty sure that before March 2025 we can fix it with OpenSSL upstream, if it is still present.)

We can absolutely fix this upstream until March 2025, but somebody needs to actually get their hands dirty and bisect the commit that starts breaking this. Now, because it only happens in `+universal` and openssl uses the `muniversal` PortGroup this admittedly isn't super simple to test, but it's also not impossible either.

I would have done that, but getting an old x86_64 VM running on Apple Silicon really doesn't seem to be simple. Maybe I should try again using VMware Fusion now that it is free for personal use. QEMU and UTM couldn't get them installed.

**Comment by fhgwright** [comment]
@neverpanic 
> > @neverpanic Well, please wait, let us verify if the issue is still there or not. (And I am pretty sure that before March 2025 we can fix it with OpenSSL upstream, if it is still present.)

TL; DR; The problem went away on its own.  I'm testing an update now; PR coming soon to remove the holdback as well as fixing the broken 10.4 build.

The build works in all cases except the "usual suspect" - `10.4 ppc +universal`.  And "build" here includes running three simple programs to verify that the nasty bug is gone.  But running the real tests is time-consuming, particularly since they're flaky in come cases, requiring multiple attempts to see whether the failures are real.

It would be instructive to know what the problem actually was, but debugging non-problems isn't a high priority.

> We can absolutely fix this upstream until March 2025, but somebody needs to actually get their hands dirty and bisect the commit that starts breaking this. Now, because it only happens in `+universal` and openssl uses the `muniversal` PortGroup this admittedly isn't super simple to test, but it's also not impossible either.

I'm not sure that the problem was strictly limited to `+universal`, and it was an attempt to check that that led me to discover that the problem is now gone.

The biggest hurdle to bisecting is that actually building each candidate as a port would be incredibly time-consuming and painful, so one first has to create some shell scripts to duplicate the MacPorts build environment well enough to test directly in the source repo (or a clone thereof).  It's even worse when patches are needed, though fortunately that isn't the case here.

If the problem were truly confined to universal builds, then the scripts would also need to duplicate the `muniversal` behavior.

> I would have done that, but getting an old x86_64 VM running on Apple Silicon really doesn't seem to be simple. Maybe I should try again using VMware Fusion now that it is free for personal use. QEMU and UTM couldn't get them installed.

Running x86_64 on Apple Silicon isn't a VM, it's an emulation.  VMware doesn't do that.  UTM does, but unfortunately in emulation mode it doesn't directly offer a Mac as a guest, in spite of touting exactly that in their advertising (though their only specific Mac example is running 9.2.1 on an emulated ppc).

**Comment by barracuda156** [comment]
@fhgwright @neverpanic This build of 3.3.1 worked for me on PowerPC with no issues. Tests either take forever or possibly freeze on some cases; I tried once and gave up on waiting since needed machine for something else.
